File: Demos/Dropwizard/src/main/java/com/groupdocs/ui/signature/service/SortedSignaturesData.java
package com.groupdocs.ui.signature.service;

import com.groupdocs.ui.signature.entity.web.SignatureDataEntity;

import java.util.ArrayList;
import java.util.List;

import static com.groupdocs.ui.signature.util.SignatureType.*;

/**
 * Class is used for sorting list of signature's data by signature type into different lists
 */
public class SortedSignaturesData {
    List<SignatureDataEntity> signaturesData;
    List<SignatureDataEntity> images = new ArrayList<>();
    List<SignatureDataEntity> texts = new ArrayList<>();
    List<SignatureDataEntity> codes = new ArrayList<>();
    List<SignatureDataEntity> stamps = new ArrayList<>();
    List<SignatureDataEntity> digital = new ArrayList<>();

    public SortedSignaturesData(List<SignatureDataEntity> signaturesData) {
        this.signaturesData = signaturesData;
    }

    public SortedSignaturesData sort() {
        for (int i = 0; i < signaturesData.size(); i++) {
            addToList(signaturesData.get(i));
        }
        return this;
    }

    private void addToList(SignatureDataEntity signatureDataEntity) {
        switch (signatureDataEntity.getSignatureType()) {
            case TEXT:
                texts.add(signatureDataEntity);
                break;
            case DIGITAL:
                digital.add(signatureDataEntity);
                break;
            case IMAGE:
            case HAND:
                images.add(signatureDataEntity);
                break;
            case STAMP:
                stamps.add(signatureDataEntity);
                break;
            case QR_CODE:
            case BAR_CODE:
                codes.add(signatureDataEntity);
                break;
            default:
                throw new IllegalArgumentException("Signature type is wrong");
        }
    }
}

File: Demos/Ktor/src/main/kotlin/com/groupdocs/ui/signature/ktor/modules/sign/SortedSignaturesData.kt
package com.groupdocs.ui.signature.ktor.modules.sign

import com.groupdocs.ui.signature.ktor.model.SignatureDataEntity
import com.groupdocs.ui.util.SignatureType.BAR_CODE
import com.groupdocs.ui.util.SignatureType.DIGITAL
import com.groupdocs.ui.util.SignatureType.HAND
import com.groupdocs.ui.util.SignatureType.IMAGE
import com.groupdocs.ui.util.SignatureType.QR_CODE
import com.groupdocs.ui.util.SignatureType.STAMP
import com.groupdocs.ui.util.SignatureType.TEXT

class SortedSignaturesData {
    var signaturesData: List<SignatureDataEntity>? = null
    var images: MutableList<SignatureDataEntity> = ArrayList<SignatureDataEntity>()
    var texts: MutableList<SignatureDataEntity> = ArrayList<SignatureDataEntity>()
    var codes: MutableList<SignatureDataEntity> = ArrayList<SignatureDataEntity>()
    var stamps: MutableList<SignatureDataEntity> = ArrayList<SignatureDataEntity>()
    var digital: MutableList<SignatureDataEntity> = ArrayList<SignatureDataEntity>()

    fun sortedSignaturesData(signaturesData: List<SignatureDataEntity>?) {
        this.signaturesData = signaturesData
    }

    fun sort(): SortedSignaturesData {
        for (i in signaturesData!!.indices) {
            addToList(signaturesData!![i])
        }
        return this
    }

    private fun addToList(signatureDataEntity: SignatureDataEntity) {
        when (signatureDataEntity.signatureType) {
            TEXT -> texts.add(signatureDataEntity)
            DIGITAL -> digital.add(signatureDataEntity)
            IMAGE, HAND -> images.add(signatureDataEntity)
            STAMP -> stamps.add(signatureDataEntity)
            QR_CODE, BAR_CODE -> codes.add(signatureDataEntity)
            else -> throw IllegalArgumentException("Signature type is wrong")
        }
    }
}